module2/projects/little_shop/requirements.md

```diff
@@ -0,0 +1,659 @@
+* follow this pattern: `GET /api/v1/<resource>`
+
+Example JSON response for the Merchant resource:
+
+```json
+{
+  "data": [
+    {
+      "id": "1",
+        "type": "merchant",
+        "attributes": {
+          "name": "Mike's Awesome Store",
+        }
+    },
+    {
+      "id": "2",
+      "type": "merchant",
+      "attributes": {
+        "name": "Store of Fate",
+      }
+    },
+    {
+      "id": "3",
+      "type": "merchant",
+      "attributes": {
+        "name": "This is the limit of my creativity",
+      }
+    }
+  ]
+}
+```
+
+For "Fetch all" endpoints that have a condition (i.e. sorted, filtered, etc.), the following parameters should be added to the request to indicate what type of condition should be included in the response:
+
+* `?sorted=age` on the merchants index to sort merchants by created_at timestamp, newest first
+* `?sorted=price` on the items index to sort items by price, cheapest first
+* `?status=returned` on the merchants index to include only merchants that have had items from an invoice returned (hint: look at merchant's invoices)
+* `?count=true` on the merchants index to include the calculated `item_count` attribute (see below)
+
+**Calculated Attribute**
+
+One of the merchant endpoints requires the ability to add a calculated count attribute for every merchant, indicating the number of items the merchant has. This calculated attribute should be added to every merchant, in the attributes section, as see below:
+
+```json
+{
+  "data": [
+    {
+      "id": "1",
+        "type": "merchant",
+        "attributes": {
+          "name": "Mike's Awesome Store",
+          "item_count": 14
+        }
+    },
+    {
+      "id": "2",
+      "type": "merchant",
+      "attributes": {
+        "name": "Store of Fate",
+        "item_count": 24
+      }
+    }
+  ]
+}
+```
+
+
+</section>
+<section class="dropdown">
+## 2. Fetch a single record
+
+This endpoint for Items and Merchants should:
+
+* render a JSON representation of the corresponding record, if found
+* follow this pattern: `GET /api/v1/<resource>/:id`
+
+Example JSON response for the Item resource:
+
+```json
+{
+  "data": {
+    "id": "1",
+    "type": "item",
+    "attributes": {
+      "name": "Super Widget",
+      "description": "A most excellent widget of the finest crafting",
+      "unit_price": 109.99
+    }
+  }
+}
+```
+
+Note that the `unit_price` is sent as numeric data, and not string data.
+
+<hr/>
+</section>
+
+<section class="dropdown">
+## 3: Create an Item/Merchant
+
+This endpoint should:
+
+* create a record and render a JSON representation of the new Item or Merchant record.
+* follow this pattern: `POST /api/v1/items` or `POST /api/v1/merchants`
+* accept the following JSON body with only the following fields:
+
+Item:
+```json
+{
+  "name": "value1",
+  "description": "value2",
+  "unit_price": 100.99,
+  "merchant_id": 14
+}
+```
+(Note that the unit price is to be sent as a numeric value, not a string.)
+Merchant
```

Comment:
"Merchant" is showing up on the same line as the previous note rather than a new line